packages/components/src/disabled/test/index.tsx

```diff
@@ -134,6 +135,45 @@ describe( 'Disabled', () => {
 		);
 	} );
 
+	it( 'should preserve input values when toggling the isDisabled prop', async () => {
```

Comment:
A few improvements for this new test:

- let's always use `user-event` instead of `fireEvent`
- let's use the `toHaveValue` and `toHaveTextContent` assertions
- let's use the `user.type` function (which takes care of simulating clicks and keyboard events in one line)

<details>
<summary>This is how we could apply these suggestions</summary>

```diff
diff --git a/packages/components/src/disabled/test/index.tsx b/packages/components/src/disabled/test/index.tsx
index 8123ac25ea..c0858d6c5f 100644
--- a/packages/components/src/disabled/test/index.tsx
+++ b/packages/components/src/disabled/test/index.tsx
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
 
 /**
  * Internal dependencies
@@ -148,30 +148,24 @@ describe( 'Disabled', () => {
 
 		const { rerender } = render( <MaybeDisable isDisabled={ false } /> );
 
-		const getInputValue = () =>
-			screen.getByRole< HTMLInputElement >( 'textbox' ).value;
+		const input = screen.getByRole( 'textbox' );
+		const contentEditable = screen.getByTitle( 'edit my content' );
 
-		const getContentEditableValue = () =>
-			screen.getByTitle( 'edit my content' ).textContent;
+		await user.type( input, 'Input field' );
+		expect( input ).toHaveValue( 'Input field' );
 
-		const input = screen.getByRole< HTMLInputElement >( 'textbox' );
-		fireEvent.change( input, { target: { value: 'This is input.' } } );
-		expect( input.value ).toBe( 'This is input.' );
-
-		const contentEditable = screen.getByTitle( 'edit my content' );
-		await user.click( contentEditable );
-		await user.keyboard( 'This is contentEditable.' );
-		expect( contentEditable.textContent ).toBe(
-			'This is contentEditable.'
-		);
+		await user.type( contentEditable, 'contentEditable field' );
+		expect( contentEditable ).toHaveTextContent( 'contentEditable field' );
 
 		rerender( <MaybeDisable isDisabled={ true } /> );
-		expect( getInputValue() ).toBe( 'This is input.' );
-		expect( getContentEditableValue() ).toBe( 'This is contentEditable.' );
+
+		expect( input ).toHaveValue( 'Input field' );
+		expect( contentEditable ).toHaveTextContent( 'contentEditable field' );
 
 		rerender( <MaybeDisable isDisabled={ false } /> );
-		expect( getInputValue() ).toBe( 'This is input.' );
-		expect( getContentEditableValue() ).toBe( 'This is contentEditable.' );
+
+		expect( input ).toHaveValue( 'Input field' );
+		expect( contentEditable ).toHaveTextContent( 'contentEditable field' );
 	} );
 
 	describe( 'Consumer', () => {
```
</details>